docs/changelog.rst

```diff
@@ -6,11 +6,75 @@ Changelog
 Starting with version 1.8.0, pybind11 releases use a `semantic versioning
 <http://semver.org>`_ policy.
 
-v2.0.2 (not yet released)
+v2.1.0 (not yet released)
 -----------------------------------------------------
 
-* Use -x flag to strip shared libraries on OSX
-  `#595 <https://github.com/pybind/pybind11/pull/595>`_.
+* pybind11 now performs function overload resolution in two phases. The first
+  phase only considers exact type matches, while the second allows for implicit
+  conversions to take place. A special ``noconvert()`` syntax can be used to
+  completely disable implicit conversions for specific arguments.
+  `#643 <https://github.com/pybind/pybind11/pull/643>`_,
+  `#634 <https://github.com/pybind/pybind11/pull/634>`_,
+  `#650 <https://github.com/pybind/pybind11/pull/650>`_.
+
+* Ability to mix ``py::args/kwargs`` with positional arguments.
+  `#611 <https://github.com/pybind/pybind11/pull/611>`_.
+
+* Improved support for C++11 unicode string and character types; added
+  extensive documentation regarding pybind11's string conversion behavior.
+  `#624 <https://github.com/pybind/pybind11/pull/624>`_,
+  `#636 <https://github.com/pybind/pybind11/pull/636>`_,
+  `#715 <https://github.com/pybind/pybind11/pull/715>`_.
+
+* pybind11 can now avoid expensive copies when converting Eigen arrays to NumPy
+  arrays (and vice versa). `#610 <https://github.com/pybind/pybind11/pull/610>`_.
+
+* The "fast path" in ``py::vectorize`` now works for any group of C and
+  F-contiguous arrays. `#610 <https://github.com/pybind/pybind11/pull/610>`_.
+
+* Added proxy objects for fast unchecked access to NumPy arrays.
```

Comment:
Very minor wording change: "Added fast, unchecked access to NumPy arrays via a proxy object."